lib/elixir/lib/macro.ex

```diff
@@ -1801,16 +1801,27 @@ defmodule Macro do
     do: {:filename.dirname(env.file), true}
 
   defp do_expand_once({:__ENV__, _, atom}, env) when is_atom(atom) do
-    env = update_in(env.versioned_vars, &maybe_escape_map/1)
-    {maybe_escape_map(env), true}
+    env =
+      Map.new(env, fn
+        # TODO handle unescapable values which are deeper in the structure
+        {key, value} when is_reference(x) or is_pid(x) or is_port(x) -> {key, nil}
+        {key, value} -> {key, value}
+      end)
```

Comment:
Unfortunately this is much more expensive. I think you can do:

```
env = put_in(env.lexical_tracker, nil)
```

and that should be it. We don't have any improper fields anywhere else. :)

If you want, you can add a test that traverses the result of Macro.expand recursively and verifies there is no improper data. :)